It can't link on aarch64-linux.

```
make[4]: Entering directory '/build/fpcbuild-3.2.0/fpcsrc/packages/fpmkunit'
/nix/store/n35z01fwkf0alzn5igg5l4phc9r8gdk6-coreutils-9.0/bin/mkdir -p units_bs/aarch64-linux
/build/fpcbuild-3.2.0/fpcsrc/compiler/ppca64 src/fpmkunit.pp  -Fu/build/fpcbuild-3.2.0/fpcsrc/rtl/units/aarch64-linux -FUunits_bs/aarch64-linux -Fu../paszlib/src -Fu../hash/src -Fi../paszlib/src  -Fi../fcl-process/src/unix -Fu../fcl-process/src -Fi../fcl-process/src/linux -Fi../fcl-process/src/dummy -Fu../libtar/src 
make[4]: Leaving directory '/build/fpcbuild-3.2.0/fpcsrc/packages/fpmkunit'
/build/fpcbuild-3.2.0/fpcsrc/compiler/ppca64 fpmake.pp  -Fu/build/fpcbuild-3.2.0/fpcsrc/packages/fpmkunit/units_bs/aarch64-linux -Fu/build/fpcbuild-3.2.0/fpcsrc/rtl/units/aarch64-linux  
/nix/store/vwsfkvn1zp4z0gznxx2r478v0rrwncpk-binutils-2.38/bin/ld: /build/fpcbuild-3.2.0/fpcsrc/rtl/units/aarch64-linux/cprt0.o: in function `_start':
(.text+0x54): undefined reference to `__libc_csu_init'
/nix/store/vwsfkvn1zp4z0gznxx2r478v0rrwncpk-binutils-2.38/bin/ld: (.text+0x58): undefined reference to `__libc_csu_init'
/nix/store/vwsfkvn1zp4z0gznxx2r478v0rrwncpk-binutils-2.38/bin/ld: (.text+0x5c): undefined reference to `__libc_csu_fini'
/nix/store/vwsfkvn1zp4z0gznxx2r478v0rrwncpk-binutils-2.38/bin/ld: (.text+0x60): undefined reference to `__libc_csu_fini'
fpmake.pp(55) Error: Error while linking
fpmake.pp(55) Fatal: There were 1 errors compiling module, stopping
Fatal: Compilation aborted
make[3]: *** [Makefile:1725: fpmake] Error 1
make[3]: Leaving directory '/build/fpcbuild-3.2.0/fpcsrc/packages'
make[2]: *** [Makefile:2681: packages_smart] Error 2
make[2]: Leaving directory '/build/fpcbuild-3.2.0/fpcsrc'
make[1]: *** [Makefile:2836: build-stamp.aarch64-linux] Error 2
make[1]: Leaving directory '/build/fpcbuild-3.2.0/fpcsrc'
make: *** [Makefile:2307: fpcsrc/build-stamp.aarch64-linux] Error 2
```

--- a/pkgs/applications/networking/p2p/transgui/default.nix
+++ b/pkgs/applications/networking/p2p/transgui/default.nix
@@ -1,30 +1,38 @@
-{ lib, stdenv, fetchFromGitHub, pkg-config, makeDesktopItem, unzip, fpc, lazarus,
-libX11, glib, gtk2, gdk-pixbuf, pango, atk, cairo, openssl }:
+{ lib, stdenv, fetchFromGitHub, pkg-config, makeDesktopItem, fetchpatch, unzip
+, fpc, lazarus, libX11, glib, gtk2, gdk-pixbuf, pango, atk, cairo, openssl }:
 
 stdenv.mkDerivation rec {
   pname = "transgui";
-  version = "5.18.0";
+  version = "unstable-2022-02-02";
 
   src = fetchFromGitHub {
     owner = "transmission-remote-gui";
     repo = "transgui";
-    rev = "v${version}";
-    sha256 = "1dyx778756zhvz5sxgdvy49p2c0x44w4nmcfd90wqrmgfknncnf5";
+    rev = "0e2c2a07c1b21b1704c0a4945a111a8aa1050a1a";
+    sha256 = "1x9wzii3q9zanpik4xc99jqsfrqch8vjmlx12jrvczxcfy51b1ba";
   };
+
+  patches = [
+    # TDDO: remove when transgui updates for transmission-daemon v3 rpc protocol
+    (fetchpatch {
+      url = "https://github.com/transmission-remote-gui/transgui/commit/9275c3fb877dd753a1940d1b900630cdc09a0cc2.patch";
+      sha256 = "0w2x7gcxp5kqczdz7ckfqhdz9hhkm62k8gcws54d6km7x9vc1023";
+    })
+  ];
 
   nativeBuildInputs = [ pkg-config unzip ];
   buildInputs = [
-    fpc lazarus stdenv.cc
-    libX11 glib gtk2 gdk-pixbuf pango atk cairo openssl
+    fpc lazarus stdenv.cc libX11 glib gtk2 gdk-pixbuf
+    pango atk cairo openssl
   ];
 
-  NIX_LDFLAGS = "
-    -L${stdenv.cc.cc.lib}/lib
-    -lX11 -lglib-2.0 -lgtk-x11-2.0 -lgdk-x11-2.0
-    -lgdk_pixbuf-2.0 -lpango-1.0 -latk-1.0 -lcairo -lc -lcrypto
-  ";
+  NIX_LDFLAGS = ''
+    -L${stdenv.cc.cc.lib}/lib -lX11 -lglib-2.0 -lgtk-x11-2.0
+    -lgdk-x11-2.0 -lgdk_pixbuf-2.0 -lpango-1.0 -latk-1.0 -lcairo
+    -lc -lcrypto
+  '';
 
-  prePatch = ''
+  postPatch = ''
     substituteInPlace restranslator.pas --replace /usr/ $out/
   '';
 
@@ -33,23 +41,19 @@ stdenv.mkDerivation rec {
     lazbuild -B transgui.lpr --lazarusdir=${lazarus}/share/lazarus
   '';
 
-  makeFlags = [
-    "FPC=fpc"
-    "PP=fpc"
-    "INSTALL_PREFIX=$(out)"
-  ];
+  makeFlags = [ "FPC=fpc" "PP=fpc" "INSTALL_PREFIX=$(out)" ];
 
   LCL_PLATFORM = "gtk2";
 
-  desktopItem = makeDesktopItem rec {
-    name = "transgui";
-    exec = name + " %U";
-    icon = name;
+  desktopItem = makeDesktopItem {
+    name = pname;
+    exec = "${pname} %U";
+    icon = pname;
     type = "Application";
     comment = meta.description;
     desktopName = "Transmission Remote GUI";
     genericName = "BitTorrent Client";
-    categories = [ "Application" "Network" "FileTransfer" "P2P" "GTK" ];
+    categories = [ "Network" "FileTransfer" "P2P" "GTK" ];
     startupNotify = true;
     mimeTypes = [ "application/x-bittorrent" "x-scheme-handler/magnet" ];
   };
@@ -64,10 +68,11 @@ stdenv.mkDerivation rec {
   '';
 
   meta = {
-    description = "A cross platform front-end for the Transmission Bit-Torrent client";
+    description = "A cross platform front-end for the Transmission BitTorrent client";
     homepage = "https://sourceforge.net/p/transgui";
     license = lib.licenses.gpl2Plus;
     maintainers = with lib.maintainers; [ ramkromberg ];
-    platforms = lib.platforms.linux;
+    mainProgram = "transgui";
+    platforms = [ "x86_64-linux" "x86_64-darwin" ];
   };
 }

